### DISCUSSION TITLE
how to merge all wall surfaces? rotor 37

### DISCUSSION THREAD

**Discussion by Mohsin-Bashir** [mdolab/dafoam#681] [General]
https://github.com/mdolab/dafoam/discussions/671#discussioncomment-10475091

I tried the following way:
1. **First** changed the boundary file, initially it was like this:
```
    blade
    {
        type            wall;
        inGroups        1(wall);
        nFaces          2670;
        startFace       121224;
    }
    shroud
    {
        type            path;
        nFaces          2826;
        startFace       123894;
    }
    hub
    {
        type            patch;
        nFaces          2826;
        startFace       126720;
    }
```
Then changed these lines to:
```

    blade
    {
        type            wall;
        inGroups        1(wall);
        nFaces          8322;
        startFace       121224;
    }
```
2. **Secondly,** I removed all lines related to hub and shroud in files from 0 and 0.org directory.

But still could not run the simulation. I am trying all this for rotor 37 tutorial case. What's the correct way of merging wall surfaces?

**Comment by benoitpaillard** [comment]
You can use OpenFOAM tool mergeSurfacePatches for that.

**Reply by Mohsin-Bashir** [reply]
Are you certain this tool works for volume meshes as well? From what I’ve seen in the documentation, it seems to only apply to surface meshes. If it does work for volumetric meshes, could you please share more details on how to use it in that context?

**Comment by Mohsin-Bashir** [comment]
I’ve found a better method to merge surfaces. Initially, I had a Fluent .msh mesh. I modified the system/createPatchDict file by first removing the lines related to the hub and shroud. Then, I updated the blade dictionary as follows:
```
    {
        // Name of new patch
        name blade;

        // Dictionary to construct new patch from
        patchInfo
        {
            type wall;
        }

        // How to construct: either from 'patches' or 'set'
        constructFrom patches;
        patches (BLADE SHROUD HUB);
    }
```
Next, run the `preprocessing.sh` file directly. This will generate the OpenFOAM mesh, with the blade, shroud, and hub surfaces merged as a single blade surface. Let me know if there is a better way to do this.

**Reply by friedenhe** [reply]
Yes, we should use createPatch to merge surfaces

**Comment by NeuroNexusAI** [comment]
To merge wall surfaces seamlessly, you should start by preparing the surfaces—cleaning and sanding them. Apply a primer to ensure better adhesion. Use joint compound or plaster to fill any gaps and smooth out the transition. After drying, sand the area to achieve a uniform surface, and then paint over it for a cohesive look.

**Reply by Mohsin-Bashir** [reply]
Nice joke

**Comment by friedenhe** [comment]
@Mohsin-Bashir, I believe @NeuroNexusAI is an AI-based robot, so please ignore its answers.
